### ISSUE TITLE
Models are recreated every time - Hugely inefficient

### ISSUE THREAD

**Issue by oxysoft** [Fannovel16/comfyui_controlnet_aux#13]
Hi, I was investigating why the nodes were so slow compared to my own code and it seems that the pipelines / models are instantiated anew on each run. I have 24GB VRAM and would prefer to keep everything loaded in and get massive speedups instead. I believe this should be possible, ComfyUI does it with their model loader.

Cheers

**Comment by Fannovel16** [comment]
Yep it is possible but currently idk how to do memory management on these models. If all models are cached in the same time then the memory usage will be huge. Do you have any idea?

**Comment by oxysoft** [comment]
Certainly, I think you could re-use the arguments that ComfyUI uses to control the memory management policy for their model loading. For me in `--highvram` mode I would be happy if it simply never unloaded the models until the node is removed. (one global instance of the model used for all instances of the same preprocessor, then deleted when none of them remain)
